src/Neo/Ledger/MemoryPool.cs

```diff
@@ -320,6 +341,49 @@ internal VerifyResult TryAdd(Transaction tx, DataCache snapshot)
             return VerifyResult.Succeed;
         }
 
+        /// <summary>
+        /// Checks whether there is no mismatch in Conflicts attributes between the current transaction
+        /// and mempooled unsorted transactions. If true, then these unsorted transactions will be added
+        /// into conflictsList.
+        /// </summary>
+        /// <param name="tx">The <see cref="Transaction"/>current transaction needs to be checked.</param>
+        /// <param name="conflictsList">The list of conflicting verified transactions that should be removed from the pool if tx fits the pool.</param>
+        /// <returns>True if transaction fits the pool, otherwise false.</returns>
+        private bool CheckConflicts(Transaction tx, out List<PoolItem> conflictsList)
+        {
+            conflictsList = new();
+            long conflictsFeeSum = 0;
+            // Step 1: check if `tx` was in Conflicts attributes of unsorted transactions.
+            if (_conflicts.TryGetValue(tx.Hash, out var conflicting))
+            {
+                foreach (var hash in conflicting)
+                {
+                    var unsortedTx = _unsortedTransactions[hash];
+                    if (unsortedTx.Tx.Signers.Select(s => s.Account).Contains(tx.Sender))
```

Comment:
If tx.sender is a signer, but is not the signer of the conflicted tx, he doesn't pay the tx so I think that we should not substract the amount

Reply:
Here the logic is the following: if transaction ‘tx’ conflicts with a set of mempooled transactions, then ‘tx’ should pay greater fee than the sum of all fees of those mempooled transactions (to be able to enter mempool). If fee of ‘tx’ is greater, then those conflicting transactions will be replaced by ‘tx’.

Here we check that tx.Sender is the signer (not the sender!) of mempooled conflicting transaction to avoid the situation when malicious account sends a lot of transactions that will conflict with ‘tx’ to prevent ‘tx’ from entering the pool. In other words, you (as a sender) should pay more than the sum fee of *signed by yourself* transactions to be able to replace all these conflicting transactions from the pool by a new one.